File: frontend/src/lib/api/coach.ts
import { API_BASE_URL } from "./config";

export async function getMembers() {
  const res = await fetch(`${API_BASE_URL}/coach/members`);
  if (!res.ok) {
    throw new Error("Failed to fetch members");
  }
  return res.json();
}

export async function getMemberDetails(userId: string) {
  const res = await fetch(`${API_BASE_URL}/coach/members/${userId}`);

  if (!res.ok) {
    throw new Error("Failed to fetch member");
  }

  return res.json();
}

export async function getMemberWorkout(userId: string) {
  const res = await fetch(
    `${API_BASE_URL}/coach/members/${userId}/workout`
  );
  if (!res.ok) {
    throw new Error("Failed to fetch workout");
  }
  return res.json();
}

export async function getMemberMeal(userId: string) {
  const res = await fetch(
    `${API_BASE_URL}/coach/members/${userId}/meal`
  );
  if (!res.ok) {
    throw new Error("Failed to fetch meal");
  }
  return res.json();
}

File: backend/src/api/coach.py
from fastapi import APIRouter
from src.config.database import db

router = APIRouter(
    prefix="/coach",
    tags=["Coach"]
)

@router.get("/members")
def get_members():
    users = list(
        db["users"].find(
            {"role": "user"},
            {"_id": 0}
        )
    )
    for user in users:
        subscription = db["subscriptions"].find_one(
            {"user_id": user["user_id"]},
            {"_id": 0}
        )
        if subscription:
            user["plan"] = subscription.get("plan", "trial")
            user["subscription_status"] = subscription.get("status", "active")
        else:
            user["plan"] = "trial"
            user["subscription_status"] = "inactive"
    return users

@router.get("/members/{user_id}")
def get_member(user_id: str):
    user = db["users"].find_one(
        {"user_id": user_id},
        {"_id": 0}
    )
    if not user:
        return {
            "success": False,
            "message": "Member not found"
        }
    subscription = db["subscriptions"].find_one(
        {"user_id": user_id},
        {"_id": 0}
    )
    return {
        "success": True,
        "user": user,
        "subscription": subscription
    }

@router.get("/members/{user_id}/workout")
def get_member_workout(user_id: str):
    workout = db["workout_plans"].find_one(
        {"user_id": user_id},
        {"_id": 0}
    )
    return workout

@router.get("/members/{user_id}/meal")
def get_member_meal(user_id: str):
    meal = db["meal_plans"].find_one(
        {"user_id": user_id},
        {"_id": 0}
    )
    return meal